include/acl_selectors.php

```diff
@@ -625,31 +631,27 @@ function _contact_link($i){ return dbesc($i['link']); }
 				intval($conv_id),
 				dbesc($search),
 				dbesc($search),
-				implode("','", $known_contacts)
+				implode("', '", $known_contacts)
 		);
-		if (dbm::is_result($r)){
+		if (dbm::is_result($r)) {
 			foreach ($r as $row) {
-				// nickname..
-				$up = parse_url($row['author-link']);
-				$nick = explode("/",$up['path']);
-				$nick = $nick[count($nick)-1];
-				$nick .= "@".$up['host'];
-				// /nickname
-				$unknow_contacts[] = array(
-					'type'    => 'c',
-					'photo'   => proxy_url($row['author-avatar'], false, PROXY_SIZE_MICRO),
-					'name'    => htmlentities($row['author-name']),
-					'id'      => '',
-					'network' => 'unknown',
-					'link'    => $row['author-link'],
-					'nick'    => htmlentities($nick),
+				$contact = get_contact_details_by_url($row['author-link'], 0);
```

Comment:
Using ```get_contact($url);``` would ensure having data for everything that is in our database or that is resolvable via "probe". For everything else you could maybe call a little function that is doing that bad hack the we had done before.

Normally the mentioned accounts had already posted stuff (otherwise a user wouldn't mention them, I think), so the probing would be a solution for the exotic case when someone is mentioned who isn't known to this instance.